packages/l/libsoundio/xmake.lua

```diff
@@ -5,12 +5,16 @@ package("libsoundio")
 
     set_urls("https://github.com/andrewrk/libsoundio/archive/$(version).tar.gz",
              "https://github.com/andrewrk/libsoundio.git")
+
     add_versions("2.0.0", "67a8fc1c9bef2b3704381bfb3fb3ce99e3952bc4fea2817729a7180fddf4a71e")
+    add_patches("2.0.0", path.join(os.scriptdir(), "patches", "2.0.0", "extern.patch"), "c1c9084012cf39b203cdf7c59d46a8e6362b6a1e7203700d77faef529d49d401")
 
     add_deps("cmake")
 
     add_includedirs("include", "include/soundio")
 
+    set_languages("c11")
```

Comment:
It only for xmake.lua. If it is a cmake package, you need to pass `/TP` and `/std:c++11` to cmake to force it to be compiled as c++11

```lua
import("package.tools.cmake").install(package, configs, {cxflags = {"/TP", "/std:c++11"}})

Reply:
Thanks. But it seems to ignore /TP as it's still compiling in C mode.
I tried /TP locally (to test my patch) and it did work.

Reply:
But stdatomic.h is missing in older vs

Reply:
may be we need patch a stdatomic.h

Reply:
stdatomic.h shouldn't be required in C++ mode

Reply:
I have used xmake.lua to port it and it works on windows. 

But it only support to enable coreaudio backend now. 

If you want to support other backends, you may need to make some improvements. But I'm going to sleep, you can try to continue to improve it on the libsoundio branch. https://github.com/xmake-io/xmake-repo/blob/libsoundio/packages/l/libsoundio/xmake.lua